src/wap_client.c

```diff
@@ -211,7 +221,8 @@ signal_have_save_ok (client_t *self)
 static void
 signal_have_start_ok (client_t *self)
 {
-    zsock_send (self->cmdpipe, "si", "START OK", 0);
+    zsock_send (self->cmdpipe, "sii", "START OK", 0,
+                wap_proto_curr_height (self->message));
 }
```

Comment:
It's C code and we can't mix hand-built and generated code without quite
clumsy effort.

Generating this plumbing would be possible though not without additional
complexity elsewhere. For now it's simpler to write these zsock_send
replies by hand. I'll come back to this at a later stage when we have more
experience with the client layers.

This code is invisible to the wallet developer. It's the work of the API
developer (delivering wap_client), in two places (setting up the protocol
commands to the server, and sending replies back to the client).




On Sun, Jan 4, 2015 at 6:11 PM, oranjuice <notifications@github.com> wrote:

> In src/wap_client.c
> <https://github.com/oranjuice/monero-0mq/pull/4#discussion-diff-22440675>:
>
> > @@ -211,7 +221,8 @@ signal_have_save_ok (client_t *self)
> >  static void
> >  signal_have_start_ok (client_t *self)
> >  {
> > -    zsock_send (self->cmdpipe, "si", "START OK", 0);
> > +    zsock_send (self->cmdpipe, "sii", "START OK", 0,
> > +                wap_proto_curr_height (self->message));
> >  }
>
> Couldn't the system figure this out on its own? If I added another field
> to be returned along with curr_height in the daemon, I would have to pass
> it here to zsock_send too.
>
> The wallet developer would have to act upon such changes in the message
> formats (extra field, etc.) in two places: one here and another in the real
> wallet code where the response is actually made use of by the wallet.
>
> I feel the same way about prepare_start_command too. If the start message
> format as predecided, is to contain a "start_height" field, shouldn't the
> underlying system be able to take care of everything (like storing it in a
> message) after a wap_client_start call?
>
> Of course, I'm not very knowledgeable about the internals, but that
> would've been simpler usage IMO, unless there is a benefit of such finer
> grain control.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oranjuice/monero-0mq/pull/4/files#r22440675>.
>